packages/vscode-extension/src/project/project.ts

```diff
@@ -183,6 +193,11 @@ export class Project implements Disposable, ProjectInterface, DeviceSessionsMana
     });
   }
 
+  public async runDependencyChecks(): Promise<void> {
+    await this.applicationContext.applicationDependencyManager.runAllDependencyChecks();
+    await this.environmentDependencyManager.runAllDependencyChecks();
```

Comment:
Shouldn't these be run in parallel?

```suggestion
    await Promise.all([
      this.applicationContext.applicationDependencyManager.runAllDependencyChecks(),
      this.environmentDependencyManager.runAllDependencyChecks()
    ])
```